packages/web-sdk/src/instrumentations/webVitalsWithAttribution/instrumentation.ts

```diff
@@ -0,0 +1,128 @@
+      this.addIfPresent(values, 'time_to_first_byte', metric.attribution.timeToFirstByte)
+
+      const context = this.buildInitialContext(metric)
+      this.addIfPresent(context, 'load_state', metric.attribution.loadState)
+
+      this.pushMeasurement(values, context)
+    })
+  }
+
+  private measureFID(): void {
+    onFID((metric) => {
+      const values = this.buildInitialValues(metric)
+      this.addIfPresent(values, 'event_time', metric.attribution.eventTime)
+
+      const context = this.buildInitialContext(metric)
+      this.addIfPresent(context, 'event_target', metric.attribution.eventTarget)
+      this.addIfPresent(context, 'event_type', metric.attribution.eventType)
+      this.addIfPresent(context, 'load_state', metric.attribution.loadState)
+
+      this.pushMeasurement(values, context)
+    })
+  }
+
+  private measureINP(): void {
+    onINP((metric) => {
+      const values = this.buildInitialValues(metric)
+      this.addIfPresent(values, 'event_time', metric.attribution.eventTime)
+
+      const context = this.buildInitialContext(metric)
+      this.addIfPresent(context, 'event_target', metric.attribution.eventTarget)
+      this.addIfPresent(context, 'event_type', metric.attribution.eventType)
+      this.addIfPresent(context, 'load_state', metric.attribution.loadState)
+
+      this.pushMeasurement(values, context)
+    })
+  }
+
+  private measureLCP(): void {
+    onLCP((metric) => {
+      const values = this.buildInitialValues(metric)
+      this.addIfPresent(values, 'element_render_delay', metric.attribution.elementRenderDelay)
+      this.addIfPresent(values, 'resource_load_delay', metric.attribution.resourceLoadDelay)
+      this.addIfPresent(values, 'resource_load_time', metric.attribution.resourceLoadTime)
+      this.addIfPresent(values, 'time_to_first_byte', metric.attribution.timeToFirstByte)
+
+      const context = this.buildInitialContext(metric)
+      this.addIfPresent(context, 'element', metric.attribution.element)
+
+      this.pushMeasurement(values, context)
+    })
+  }
+
+  private measureTTFB(): void {
+    onTTFB((metric) => {
+      const values = this.buildInitialValues(metric)
+      this.addIfPresent(values, 'dns_time', metric.attribution.dnsTime)
+      this.addIfPresent(values, 'connection_time', metric.attribution.connectionTime)
+      this.addIfPresent(values, 'request_time', metric.attribution.requestTime)
+      this.addIfPresent(values, 'waiting_time', metric.attribution.waitingTime)
+
+      const context = this.buildInitialContext(metric)
+
+      this.pushMeasurement(values, context)
+    })
+  }
+
+  private buildInitialValues(metric: Metric): Values {
+    const indicator = metric.name.toLowerCase()
+    return { [indicator]: metric.value }
+  }
+
+  private buildInitialContext(metric: Metric): Context {
+    return { rating: metric.rating }
+  }
+
+  private pushMeasurement(values: Values, context: Context): void {
+    const type = 'web-vitals'
```

Comment:
we are sending different information with the same type and I don't know how confusing this can get. I like the idea where the "type" is the name of the metric, I feel like things would be a little more organized that way

In the end, I preferred to go with an API that is compatible with the light version of the web vitals instrumentation, to facilitate the migration from one instrumentation to the other.